src/fides/api/service/messaging/messaging_crud_service.py

```diff
@@ -81,31 +87,261 @@ def get_messaging_config_by_key(db: Session, key: FidesKey) -> MessagingConfigRe
 
 
-def get_messaging_template_by_key(db: Session, key: str) -> Optional[MessagingTemplate]:
-    template = MessagingTemplate.get_by(db, field="key", value=key)
+def get_messaging_template_by_type(
+    db: Session, template_type: str
+) -> Optional[MessagingTemplate]:
+    template = MessagingTemplate.get_by(db, field="type", value=template_type)
 
     # If no template is found in the database, use the default
-    if not template and key in DEFAULT_MESSAGING_TEMPLATES:
-        content = DEFAULT_MESSAGING_TEMPLATES[key]["content"]
-        template = MessagingTemplate(key=key, content=content)
+    if not template and template_type in DEFAULT_MESSAGING_TEMPLATES:
+        content = DEFAULT_MESSAGING_TEMPLATES[template_type]["content"]
+        template = MessagingTemplate(type=template_type, content=content)
 
     return template
+
+
+def _validate_overlapping_templates(
+    db: Session,
+    template_type: str,
+    new_property_ids: Optional[List[str]],
+    is_enabled: bool,
+    update_id: Optional[str],
+) -> None:
+    """
+    Validates that only one enabled templates are unique by template type and property.
+
+    The following illustrates what template combinations is and isn't allowed to co-exist in the DB:
+
+    Valid:
+    template_1: {"is_enabled": True, "type": "subject_identity_verification", properties: ["FDS-13454"]}
+    template_2: {"is_enabled": False, "type": "subject_identity_verification", properties: ["FDS-13454"]}
+
+    Valid:
+    template_1: {"is_enabled": True, "type": "subject_identity_verification", properties: ["FDS-13454"]}
+    template_2: {"is_enabled": True, "type": "subject_identity_verification", properties: ["FDS-00000"]}
+
+    Invalid:
+    template_1: {"is_enabled": True, "type": "subject_identity_verification", properties: ["FDS-13454"]}
+    template_2: {"is_enabled": True, "type": "subject_identity_verification", properties: ["FDS-13454", "FDS-00000"]}
+    """
+    # We don't care about templates that are disabled or have no properties
+    if is_enabled is False or new_property_ids is None:
+        return
+    possible_overlapping_templates = (
+        db.query(MessagingTemplate)
+        .filter_by(
+            is_enabled=True,
+            type=template_type,
+        )
+        .all()
+    )
+
+    if not possible_overlapping_templates:
+        return
+
+    # If we're updating the only possible overlap, we allow this
+    if update_id and len(possible_overlapping_templates) == 1:
+        if update_id == possible_overlapping_templates[0].id:
+            return
+
+    # Otherwise, we need to check whether properties will overlap
+    for db_template in possible_overlapping_templates:
+        for db_property in db_template.properties:
+            if db_property.id in new_property_ids:
+                raise MessagingConfigValidationException(
+                    f"There is already an enabled messaging template with template type {template_type} and property {db_property.id}"
```

Comment:
What if we are updating Template A.  Say there are two possible overlapping templates, Template A and Template B. Seems like Template A might fail here, because it is not excluded in the overlapping templates. Do we need to exclude `update_id` from possible_overlapping_templates?

Reply:
yes we actually check for the `update_id` matching one of the `possible_overlapping_templates` just above this code block. 

I'll add a unit test for this to be sure

Reply:
I thought this was failing for me on "update" maybe, not "create" but I'll double check on the next round

Reply:
oh hm, let me know what you see!

Reply:
ah okay yes, I think the bug I described above is still here - you have: `if update_id and len(possible_overlapping_templates) == 1` but what if there were two overlapping_templates?

Reply:
I misread that, thanks!